code/game/objects/structures/headpike.dm

```diff
@@ -29,13 +29,13 @@
 	return ..()
 
 /obj/structure/headpike/CheckParts(list/parts_list)
-	victim = locate() in parts_list
+	victim = locate(/obj/item/bodypart/head) in parts_list
 	if(!victim) //likely a mapspawned one
 		victim = new(src)
 		victim.real_name = random_unique_name(prob(50))
-	spear = locate(speartype) in parts_list
+	spear = locate(/obj/item/spear) in parts_list
 	if(!spear)
-		spear = new speartype(src)
+		spear = new /obj/item/spear()
```

Comment:
this is wrong, you're causing the different variants of headpikes to drop the wrong object now.

L9 - speartype shouldn't be null.

Reply:
better? I tested it on a local server and it gave me the correct spear back once i deconstructed it